docs/understanding-docker.md

```diff
@@ -17,274 +17,287 @@ weight = -90
-pull images down and use them to build containers. You can [sign up for a storage plan
-here](https://www.docker.com/pricing).
+A Docker registry stores Docker images. After you build a Docker image, you
+can *push* it to a public registry such as [Docker Hub](https://hub.docker.com)
+or to a private registry running behind your firewall. You can also search for
+existing images and pull them from the registry to a host.
+
+[Docker Hub](https://hub.docker.com) is a hosted Docker registry which provides
+both public and private storage locations (repositories) for images. Images
+stored in a public repository on Docker Hub can be searched for and downloaded
+by any user. Private repositories are excluded from search results and only that
+repository's users can pull images down and use them to build containers. You can
+[sign up for a repository plan here](https://www.docker.com/pricing).
 
 ### How does a container work?
-A container consists of an operating system, user-added files, and meta-data. As
-we've seen, each container is built from an image. That image tells Docker
-what the container holds, what process to run when the container is launched, and
-a variety of other configuration data. The Docker image is read-only. When
-Docker runs a container from an image, it adds a read-write layer on top of the
-image (using a union file system as we saw earlier) in which your application can
-then run.
+A container uses the host machine's Linux kernel, and consists of any extra files
+you add when the image is created, along with metadata associated with the container
+at creation or when the container is started. Each container is built from an image.
+The image defines the container's contents, which process to run when the container
+is launched, and a variety of other configuration details. The Docker image is
+read-only. When Docker runs a container from an image, it adds a read-write layer
+on top of the image (using a UnionFS as we saw earlier) in which your application
+runs.
 
 ### What happens when you run a container?
-Either by using the `docker` binary or via the API, the Docker client tells the Docker
-daemon to run a container.
+When you use the `docker run` CLI command or the equivalent API, the Docker Engine
+client instructs the Docker daemon to run a container. This example tells the
+Docker daemon to run a container using the `ubuntu` Docker image, to remain in
+the foreground in interactive mode (`-i`), and to run the `/bin/bash` command.
 
     $ docker run -i -t ubuntu /bin/bash
 
-The Docker Engine client is launched using the `docker` binary with the `run` option
-running a new container. The bare minimum the Docker client needs to tell the
-Docker daemon to run the container is:
 
-* What Docker image to build the container from, for example, `ubuntu`
-* The command you want to run inside the container when it is launched,
-for example,`/bin/bash`
+When you run this command, Docker Engine does the following:
+
```

Comment:
I was under the impression that they did happen in this order. How can I verify it?